dex/encode/encode.go

```diff
@@ -125,7 +126,14 @@ func ExtractPushes(b []byte, preAlloc ...int) ([][]byte, error) {
 				return nil, fmt.Errorf("2 bytes not available for uint16 data length")
 			}
 			l = int(IntCoder.Uint16(b[:2]))
-			b = b[2:]
+			if l < 255 {
+				// This indicates it's really a uint32 capped at 0x00fe_ffff, and
+				// we are looking at the top two bytes.
+				l = int(IntCoder.Uint32(b[:4]))
```

Comment:
This works because any length under 255 would have just been encoded as the first byte correct?

There's a check for if `len(b) < 2` before this to make sure there's enough data left for the two byte uint16. Maybe another check for the two more bytes of the uint32 is prudent?

Also the error message "2 bytes not available for uint16 data length" may be wrong because may be uint32